pkg/gitserver/gitserver.go

```diff
@@ -184,12 +197,22 @@ func NewEnviromentConfig() (*Config, error) {
 		return nil, fmt.Errorf("only one of REQUIRE_SERVER_AUTH or REQUIRE_GIT_AUTH may be specified")
 	}
 
+	config.WatchBuildConfigs = os.Getenv("BUILDCONFIG_AUTOSTART") == "yes"
```

Comment:
Why support this AND auto linking?  Why is this a boolean and not a "type"?  What other things would we want to add in the future?